src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,176 @@
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    mockConvoStore = Mockito.mock(ConversationStore.class);
+    mockMessageStore = Mockito.mock(MessageStore.class);
+
+    User fakeUser =
+        new User(
+            UUID.randomUUID(),
+            "test username",
+            "$2a$10$bBiLUAVmUFK6Iwg5rmpBUOIBW6rIMhU1eKfi3KR60V9UXaYTwPfHy",
+            Instant.now());
+
+    mockUserStore.addUser(fakeUser);
+
+    adminServlet.setUserStore(mockUserStore);
+    adminServlet.setConversationStore(mockConvoStore);
+    adminServlet.setMessageStore(mockMessageStore);
+
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/admin.jsp"))
+        .thenReturn(mockRequestDispatcher);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    adminServlet.doGet(mockRequest, mockResponse);
+    mockRequest.getParameter("username");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testGetStats() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn(null);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    List<Conversation> fakeConversationList = new ArrayList<>();
+    Conversation fakeConversation =
+        new Conversation(fakeConversationId, UUID.randomUUID(), "test_conversation", Instant.now());
+    
+    fakeConversationList.add(fakeConversation);
+
+    Mockito.when(mockConvoStore.getAllConversations())
+        .thenReturn(fakeConversationList);
+
+    List<Message> fakeMessageList = new ArrayList<>();
+    fakeMessageList.add(
+        new Message(
+            UUID.randomUUID(),
+            fakeConversationId,
+            UUID.randomUUID(),
+            "test message",
+            Instant.now()));
+    Mockito.when(mockMessageStore.getAllMessages())
+        .thenReturn(fakeMessageList);
+
+    List totalUsers = mockUserStore.getUsers();
+    List totalMessages = fakeMessageList;
+    List totalConvos = mockConvoStore.getAllConversations();
+
+    Mockito.when(mockUserStore.getUsers()).thenReturn(totalUsers);
+    Mockito.when(mockMessageStore.getAllMessages()).thenReturn(totalMessages);
+    Mockito.when(mockConvoStore.getAllConversations()).thenReturn(totalConvos);
+
+    int numTotalUsers = totalUsers.size();
+    int numTotalMessages = totalMessages.size();
+    int numTotalConvos = totalConvos.size(); 
+
+    HttpSession mockSession = Mockito.mock(HttpSession.class);
+    
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    adminServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockSession).setAttribute("numUsers", numTotalUsers);
+    Mockito.verify(mockSession).setAttribute("numMessages", numTotalMessages);
+    Mockito.verify(mockSession).setAttribute("numConvos", numTotalConvos);
+
+    Mockito.verify(mockResponse).sendRedirect("/admin");
```

Comment:
Can you clarify why this redirects to the admin page?

Reply:
This should redirect to the admin page because all the method does is update the admin stats so there's no real reason to go anywhere else.